tests/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,334 @@
+
+	if err == nil {
+		s.T().Logf("The helm chart: %s is uninstalled", s.chart)
+	}
+
+	err = s.helmClient.Cleanup()
+
+	s.Assert().NoError(err, "Failed to clean up directories used by helm client")
+
+	s.T().Logf("Deleting namespace: %s...", s.namespace)
+	err = s.k8SClient.DeleteNamespace(s.ctx, s.namespace)
+	s.Assert().NoError(err, "Cannot delete namespace")
+	if err == nil {
+		s.T().Logf("Namespace: %q deleted", s.namespace)
+	}
+
+	s.T().Log("Test setup teardown completed.")
+}
+
+func (s *Suite) TestDCGMExporter() {
+	s.CheckingPreRequisiteDCGMExporterIsUpAndRunning()
+
+	s.T().Run("Create workload pod", func(t *testing.T) {
+		t.Log("Creating a workload pod...")
+
+		var err error
+
+		s.workloadPod, err = s.k8SClient.CreatePod(s.ctx,
+			s.namespace,
+			s.labels,
+			workloadPodName,
+			workloadContainerName,
+			workloadImage,
+		)
+
+		require.NoError(t, err, "Cannot create workload pod")
+
+		require.Eventuallyf(t, func() bool {
+			isReady, err := s.k8SClient.CheckPodCondition(s.ctx, s.namespace, s.workloadPod.Name, corev1.PodScheduled)
+			assert.NoErrorf(t, err, "Cannot get pod status: %v", err)
+			return isReady
+		}, 15*time.Minute, 5*time.Second, "Failed to create pod: %s", s.workloadPod.Name)
+
+		t.Log("The workload was created.")
+	})
+
+	s.T().Run("Verify metrics", func(t *testing.T) {
+		require.EventuallyWithT(t, func(c *assert.CollectT) {
+			metrics, err := s.k8SClient.DoHttpRequest(s.ctx,
+				s.namespace,
+				s.dcgmExpPod.Name,
+				dcgmExporterPort,
+				"metrics")
+			assert.NoError(c, err, "Failed to make request to the /metrics endpoint")
+			assert.NotEmpty(c, metrics)
+
+			var parser expfmt.TextParser
+			metricFamilies, err := parser.TextToMetricFamilies(bytes.NewReader(metrics))
+			require.NoError(c, err)
+			assert.Greater(c, len(metricFamilies), 0)
+
+			for _, metricFamily := range metricFamilies {
+				assert.NotNil(c, metricFamily)
+
+				metrics := metricFamily.GetMetric()
+				assert.NotNil(c, metrics)
+
+				// Each metric must have namespace, pod and container labels
+				for _, metric := range metrics {
+					var actualLabels []string
+					for _, label := range metric.Label {
+						labelName := pointer.StringDeref(label.Name, "")
+						if slices.Contains(expectedLabels, labelName) {
+							actualLabels = append(actualLabels, labelName)
+							assert.NotNil(c, label.Value)
+							assert.NotEmptyf(c,
+								pointer.StringDeref(label.Value, ""),
+								"The %s metric contains a label named %q label with empty value.",
+								pointer.StringDeref(metricFamily.Name, ""),
+								labelName,
+							)
+						}
+					}
+					assert.Equalf(c, len(expectedLabels), len(actualLabels),
+						"Metric %s doesn't contains expected labels: %v, actual labels: %v",
+						pointer.StringDeref(metricFamily.Name, ""), expectedLabels, metric.Label)
+				}
+			}
+		}, 60*time.Second, 30*time.Second)
+	})
+}
+
+func (s *Suite) CheckingPreRequisiteDCGMExporterIsUpAndRunning() {
```

Comment:
Thanks for moving this to a seperate function.

A Concise name would be better like `DCGMExporterPrechecks`